### PR TITLE
Backport: Templates perf: resolve patterns server side

### PR DESCRIPTION
**Pull request by ellatrix**
Trac ticket: https://core.trac.wordpress.org/ticket/61228

To test: check the patterns endpoint and search for `wp:pattern`. There should be 0 results.

Screenshot 2024-05-16 at 11 23 14

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/blocks.php [hunk 1]
```diff
@@ -1344,6 +1344,60 @@ function traverse_and_serialize_block( $block, $pre_callback = null, $post_callb
 	);
 }
 
+function replace_pattern_blocks( $blocks, &$inner_content = null ) {
```
Comment (mukeshpanchal27) [inline src/wp-includes/blocks.php:1347]: The function documentation is missing.

Comment (swissspidy) [inline src/wp-includes/blocks.php:1347]: The second parameter is not being used; remove it.

Comment (swissspidy) [reply src/wp-includes/blocks.php:1347]: The function name is also rather generic

**Review by swissspidy — COMMENTED** [review]
Unit tests would be very beneficial here, there is a lot of complexity.

**Comment by ellatrix** [comment]
Closing and reopening since I didn't make this PR from my fork.

**Pull request closed without merging**